### PR TITLE
Enable building torchvision with ffmpeg regardless of version on Linux

### PR DESCRIPTION
**Pull request by prabhat00155**
torchvision now works fine with latest ffmpeg(4.3) on Linux, hence this check is not needed.

### CODE REVIEW

**Review by datumbox — APPROVED** [review]
@prabhat00155 I don't have context why the version check is removed but I'm sure you have your reasons. Assessing only the PR for what it's supposed to be doing, LGTM.

Accepting to unblock your work.

File: test/test_video_reader.py
Comment (prabhat00155) [reply test/test_video_reader.py:1237]: I am skipping this here as that's what I see being done in other tests(for example: https://github.com/pytorch/vision/blob/719e1203438d40a62cbb3bb67dc58c8eadd615ad/test/test_video_reader.py#L426). This has been documented here: https://github.com/pytorch/vision/blob/719e1203438d40a62cbb3bb67dc58c8eadd615ad/test/common_utils.py#L24.
It would be an unnecessary task to remove all these from the unit tests and add the check in `setup.py`, given this should ideally be a temporary thing.

**Comment by prabhat00155** [comment]
> @prabhat00155 I don't have context why the version check is removed but I'm sure you have your reasons. Assessing only the PR for what it's supposed to be doing, LGTM.
> 
> Accepting to unblock your work.

Thanks @datumbox! This check was introduced as we used to support ffmpeg 4.2 and there were a lot of issues with 4.3. @bjuncek did some work making torchvision work with ffmpeg 4.3 and now it works fine on Linux, hence I am removing the check.

File: test/test_video_reader.py
Comment (NicolasHug) [reply test/test_video_reader.py:1237]: The problem is that users with 3.9 will believe that the build went fine, but in reality they will get a segfault when executing the code.

I think we should prevent them from building with ffmpeg support if they have 3.9, since it will not work.

Comment (NicolasHug) [reply test/test_video_reader.py:1237]: Basically my reasoning is that we're checking for `sys.platform` in `setup.py`, so it seems like we could follow the same logic and check for the python version as well. In both cases the goal is to prevent the user from having an error either at the build, or during execution. Is this not the case?

Comment (prabhat00155) [reply test/test_video_reader.py:1237]: I completely concur with your opinion. And we should definitely not build torchvision with ffmpeg if that crashes. That should be a separate PR however. I'll look into the seg fault on Python 3.9, and if there is no quick fix, I'll send another PR with what you suggested.

Comment (NicolasHug) [reply test/test_video_reader.py:1237]: > That should be a separate PR however

I'm sorry to insist but I believe the changes above in this PR are introducing a regression. Before this PR, these tests were properly skipped because of the

`@pytest.mark.skipif(_HAS_VIDEO_OPT is False, reason="Didn't compile with ffmpeg")`

decorator at the very beginning.

Which means that the 3.9 machines weren't build with ffmpeg, which is good because it doesn't work.

Now with the removal of the check in `setup.py` of this PR, the 3.9 CI starts being built with ffmpeg support, which leads to the segfault. So this suggests that the segfault is caused by the change in this PR, right? So it seems like we should prevent this from happening on users machines as well, not just skip a failing test.

Perhaps I'm missing something, so I won't insist more. If we're keeping the `@PY39_SKIP` decorator, let's please add a comment like `#FIXME: remove skip when issue <the issue number> is resolved`, so that we remember to remove them, and so that we can gain context faster when reading the code

Comment (NicolasHug) [reply test/test_video_reader.py:1237]: As you can see in https://app.circleci.com/pipelines/github/pytorch/vision/10412/workflows/f712fdb0-08c2-4924-906b-eb1c82e79a36/jobs/790266 (current tests on the main branch), these tests are skipped on linux 3.9 because torchvision isn't built with ffmpeg.

This PR allows linux 3.9 to build with ffmpeg, which I believe is a regression.

Comment (prabhat00155) [reply test/test_video_reader.py:1237]: The checks I have removed has nothing to do with python version. It makes sure that we use ffmpeg 4.2 and skip ffmpeg 4.3. Now, with ffmpeg 4.3, there seems to be seg fault in some cases with Python 3.9, which I haven't seen on my local setup.
```
(py39_video1) prabhatroy@dev-st-p3dn24xlarge-1:/fsx/users/prabhatroy/repos/vision$ python --version
Python 3.9.6
(py39_video1) prabhatroy@dev-st-p3dn24xlarge-1:/fsx/users/prabhatroy/repos/vision$ conda list | grep ffmpeg
ffmpeg                    4.3.2                hca11adc_0    conda-forge
(py39_video1) prabhatroy@dev-st-p3dn24xlarge-1:/fsx/users/prabhatroy/repos/vision$ uname 
Linux
(py39_video1) prabhatroy@dev-st-p3dn24xlarge-1:/fsx/users/prabhatroy/repos/vision$ pytest -v test/test_video_reader.py -k test_audio_present_pts
==================================================================================================================== test session starts =====================================================================================================================
platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /fsx/users/prabhatroy/conda/envs/py39_video1/bin/python
cachedir: .pytest_cache
rootdir: /fsx/users/prabhatroy/repos/vision, configfile: pytest.ini
collected 21 items / 20 deselected / 1 selected                                                                                                                                                                                                              

test/test_video_reader.py::TestVideoReader::test_audio_present_pts PASSED                                                                                                                                                                              [100%]

====================================================================================================================== warnings summary ======================================================================================================================
test/test_video_reader.py::TestVideoReader::test_audio_present_pts
  /fsx/users/prabhatroy/conda/envs/py39_video1/lib/python3.9/site-packages/torchvision-0.11.0a0+719e120-py3.9-linux-x86_64.egg/torchvision/io/_video_opt.py:454: UserWarning: The pts_unit 'pts' gives wrong results and will be removed in a follow-up version. Please use pts_unit 'sec'.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================================================================== 1 passed, 20 deselected, 1 warning in 4.04s ========================================================================================================= 
```
Anyways, I'll fix that before merging this PR.

Comment (NicolasHug) [reply test/test_video_reader.py:1237]: > The checks I have removed has nothing to do with python version. It makes sure that we use ffmpeg 4.2 and skip ffmpeg 4.3

Indeed this wasn't the original intent of the check, but removing this check did have an indirect impact on the 3.9 linux CI builds, so they are related in an indirect way, at least when it comes to the CI machines.

I would assume that only ffmpeg 4.3 was available on the linux 3.9 CI machines, which is why torchvision wasn't built with ffmpeg prior to removing the check.

Sorry, this looks tricky :/
Have you tried sshing into the CI machines to check whether the ffmpeg version is 4.3.2 from the same `conda-forge` channel?

Comment (prabhat00155) [reply test/test_video_reader.py:1237]: I did try to ssh into the machine and it looks like it's crashing in sws_scale() which is an ffmpeg function. This would need more debugging. In the meantime, I have created a [PR](https://github.com/pytorch/vision/pull/4417) to skip building torchvision with ffmpeg for python 3.9 and removed PY39_SKIP from unit tests.

File: setup.py
Comment (NicolasHug) [inline setup.py:364]: Nit suggestion below. Also, should we simplify the `except` statement to only catch a `CalledProcessError` below?

```suggestion
            subprocess.check_output(["ffmpeg", "-version"])
```

**Review by NicolasHug — APPROVED** [review]
Thanks @prabhat00155 , this LGTM now that https://github.com/pytorch/vision/pull/4417 is merged!

File: setup.py
Comment (prabhat00155) [reply setup.py]: Makes sense.

**Pull request merged**
